app-modules/service-management/tests/ServiceRequest/CreateServiceRequestTest.php

```diff
@@ -464,11 +467,93 @@
         ])
         ->create();
 
-    $serviceRequest = ServiceRequest::factory()->create([
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
         'priority_id' => ServiceRequestPriority::factory()->create([
             'type_id' => $serviceRequestTypesWithManager->getKey(),
         ])->getKey(),
-    ]);
+    ]));
+
+    livewire(CreateServiceRequest::class)
+        ->fillForm($request->toArray())
+        ->fillForm([
+            'respondent_id' => Contact::factory()->create()->getKey(),
+        ])
+        ->call('create');
+
+    $serviceRequest = ServiceRequest::first();
 
     expect($serviceRequest->assignments()->first())->user->id->toBe($user->getKey());
 });
+
+test('check round robin assignment', function () {
+    User::factory()->licensed(LicenseType::cases())->create();
+    $team = Team::factory()->create();
+
+    $serviceRequestTypesWithManager = ServiceRequestType::factory()
+        ->hasAttached(
+            factory: $team,
+            relationship: 'managers'
+        )
+        ->state([
+            'assignment_type' => ServiceRequestTypeAssignmentTypes::RoundRobin,
+        ])
+        ->create();
+
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
+        'priority_id' => ServiceRequestPriority::factory()->create([
+            'type_id' => $serviceRequestTypesWithManager->getKey(),
+        ])->getKey(),
+    ]));
+    $users = User::orderBy('name')->orderBy('id')->get();
+
+    foreach ($users as $user) {
+        $user->grantLicense(LicenseType::RecruitmentCrm);
+        $user->teams()->attach($team);
+        $user->givePermissionTo('service_request.view-any');
+        $user->givePermissionTo('service_request.create');
+        $user->refresh();
+        actingAs($user);
```

Comment:
You don't need to setup the Users with access or permissions or anything. The creation can and should be done probably from like a single super admin User. Since the assignment has nothing to do with whether or not the User is the one logged in.